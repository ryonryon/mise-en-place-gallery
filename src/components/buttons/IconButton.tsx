import * as React from "react";
import styled from "@emotion/styled";

const IconStyledButton = styled.div`
  display: inline-block;
  margin: 16px;
  height: 24px;

  img {
    height: 100%;
    width: 100%;
  }

  &:hover {
    cursor: pointer;
  }
`;

interface Props extends React.HTMLAttributes<HTMLDivElement> {
  img: string;
}

const IconButton = ({ img, ...rest }: Props) => (
  <IconStyledButton {...rest}>
    <img src={img} alt="icon" />
  </IconStyledButton>
);

export default IconButton;
