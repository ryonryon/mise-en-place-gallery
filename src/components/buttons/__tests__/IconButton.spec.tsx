import * as React from "react";
import * as renderer from "react-test-renderer";

import IconButton from "../IconButton";

describe("IconButton test", () => {
  it("tests IconButton renders correctly ", () => {
    const iconButton = renderer.create(<IconButton img="xxx.png" />).toJSON();
    expect(iconButton).toMatchSnapshot();
  });
});
