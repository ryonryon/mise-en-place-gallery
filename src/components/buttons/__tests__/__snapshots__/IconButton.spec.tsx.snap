// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton test tests IconButton renders correctly  1`] = `
<div
  className="css-2ggvbt"
>
  <img
    alt="icon"
    src="xxx.png"
  />
</div>
`;
